home-assistant-plugin/custom_components/xknx/__init__.py

```diff
@@ -156,26 +154,23 @@ async def async_setup(hass, config):
     for platform_config, device_type in KNX_CONFIG_PLATFORM_MAPPING.items():
         if platform_config in config[DOMAIN]:
             for device_config in config[DOMAIN][platform_config]:
-                hass.data[DATA_XKNX].xknx.devices.add(
-                    create_knx_device(
-                        hass, device_type, hass.data[DATA_XKNX].xknx, device_config
-                    )
+                create_knx_device(
+                    hass, device_type, hass.data[DATA_XKNX].xknx, device_config
                 )
 
-    for component, discovery_type in (
-        ("switch", "Switch"),
-        ("climate", "Climate"),
-        ("cover", "Cover"),
-        ("light", "Light"),
-        ("sensor", "Sensor"),
-        ("binary_sensor", "BinarySensor"),
-        ("scene", "Scene"),
-        ("notify", "Notification"),
+    for platform in (
+        "switch",
+        "climate",
+        "cover",
+        "light",
+        "sensor",
+        "binary_sensor",
+        "scene",
+        "notify",
     ):
-        found_devices = _get_devices(hass, discovery_type)
```

Comment:
now a `discovery.async_load_platform` Task is started for every platform we support. It is not filtered for only configured / used platforms. Is this a problem?

Reply:
No, the filtering happens directly in the platform itself.